distributed/dashboard/worker_html.py

```diff
@@ -1,23 +1,70 @@
+import logging
 from .utils import RequestHandler, redirect
 
 
 class _PrometheusCollector(object):
-    def __init__(self, server, prometheus_client):
-        self.server = server
+    def __init__(self, server):
+        self.worker = server
+        self.logger = logging.getLogger("distributed.dask_worker")
+        self.crick_available = True
+        try:
+            import crick  # noqa: F401
+        except ImportError:
+            self.crick_available = False
+            self.logger.info("Digest-based metrics require crick to be installed")
 
     def collect(self):
-        # add your metrics here:
-        #
-        # 1. remove the following lines
-        while False:
-            yield None
-        #
-        # 2. yield your metrics
-        #     yield prometheus_client.core.GaugeMetricFamily(
-        #         'dask_worker_connections',
-        #         'Number of connections currently open.',
-        #         value=???,
-        #     )
+        from prometheus_client.core import GaugeMetricFamily
```

Comment:
Switching to importing here seems reasonable, but does this mean there is now a stray unused import somewhere?

Reply:
Previously the import was done at the init of PrometheusHandler when the PrometheusCollector is registered in the registry.
All imports which are there are currently still required in the PrometheusHandler, which was the only point the changed _PrometheusCollector class is used.
So there is no dangling import for prometheus_client anywhere in my opinion.